modules/docs/markdown/06-guides/smithy-build-config.md

```diff
@@ -0,0 +1,47 @@
+---
+sidebar_label: Smithy build config
+title: Smithy Build Configuration
+---
+
+## Introduction
+
+Smithy provides the ability to configure the Smithy build and output by a [smithy-build configuration file](https://smithy.io/2.0/guides/smithy-build-json.html#smithy-build-json). As smithy4s uses its own build logic, it generally loads its configuration from elsewhere. However, limited support for build customization using a Smithy build configuration file is available. In particular, the [OpenAPI plugin](https://smithy.io/2.0/guides/model-translations/converting-to-openapi.html) can be used to customize the OpenAPI generation.
```

Comment:
> Does this place make sense for docs, or is there any existing section it could be added to? Nothing seemed to really fit.

Yeah I think it's fine. The `Guides` section is  a bit of a misc anyway. 

> Would it make sense to mention the smithy-build JSON generation capabilities here as well? 

I think so ? Even if we end up repeating information